subsys/bluetooth/audio/Kconfig.bap

```diff
@@ -286,5 +287,8 @@ config BT_BAP_DEBUG_STREAM_SEQ_NUM
 	  the Bluetooth Audio functionality. This will provide a warning if the application
 	  provides unexpected sequence numbers.
 
+config BT_BAP_BASE
```

Comment:
Is it really needed to have a dedicated Kconfig value for this? Unused code won't be compiled

Reply:
> Is it really needed to have a dedicated Kconfig value for this? Unused code won't be compiled

Indeed. 

The promptless Kconfig option here was mostly to guard when we guard the inclusion of the .c file in `CMakeLists.txt`:
```
zephyr_library_sources_ifdef(CONFIG_BT_BAP_BASE bap_base.c)
```
Similar to how we use `BT_BAP_STREAM`. 

We can replace it with 
```
if (CONFIG_BT_BAP_BROADCAST_SINK OR CONFIG_BT_BAP_BROADCAST_ASSISTANT OR CONFIG_BT_BAP_SCAN_DELEGATOR)
  zephyr_library_sources(bap_base.c)
endif()
```

If you think that's better? 

Or do you think we should just always add it as 
```
zephyr_library_sources(bap_base.c)
```
?

Reply:
> Or do you think we should just always add it as
> 
> ```
> zephyr_library_sources(bap_base.c)
> ```
> 
> ?

Yes, that's what I was thinking of as functions that are not called will be discarded.
But to be honest, if other files are guarded the same way, let's stick to it